[CI] RandomizedTimeSeriesIT testRateGroupBySubset failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12152 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/5kohovv44exby)
- [elasticsearch-periodic-platform-support #12152 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/wxmnhfq4zzbku)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT.testRateGroupBySubset" -Dtests.seed=F17056563EAE9C0F -Dtests.locale=br-FR -Dtests.timezone=Africa/Monrovia -Druntime.java=26
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRateGroupBySubset'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed. Agg: INCREASE | Failed windows: 1 windows(0.05482456140350877%):
ROWS: 1824|Failed for row:
[0, null, null, null, null, null, 2025-07-31T09:26:50.000Z, dsiNmvQLELiPYTd, EtHOxkBWCYsFFJp, null, null, GLVRPTDcogFZiTS, 891, tha, null, PZtwSWjzRTfr, IBRyi, MwgXmEhGpvtHVubygLY, IseJaeBYVDTlqta, null, 20, 399, AFIeE, GOHlFEQKblAl, 229, dKmcXQylfTpr, null, 174, null, 315, null, 156, 939, 938, null, 6af9:d7a8:ffff:ffff:85ab:fab:ffff:ffff, GeyuwurHpYbJShvrM, XsudpQHO, dEEBULidQzIIauWCD, BhrFMCVQnOizowzzu, null]
Wanted: RateStats[count=1, max=RateRange[lower=0.0, upper=0.0], avg=RateRange[lower=0.0, upper=0.0], min=RateRange[lower=0.0, upper=0.0], sum=RateRange[lower=0.0, upper=0.0]]
Exception: 
Expected: <1L>
     but: was <0L>
Row times and values:
	TS:2025-07-31T09:26:50Z=898.0
```

**Issue Reasons:**
- [9.3] 2 consecutive failures in test testRateGroupBySubset
- [9.3] 2 consecutive failures in step almalinux-8-aarch64_checkpart3_platform-support-arm
- [9.3] 2 failures in test testRateGroupBySubset (6.5% fail rate in 31 executions)
- [9.3] 2 failures in step almalinux-8-aarch64_checkpart3_platform-support-arm (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -331,12 +331,13 @@ public class RandomizedTimeSeriesIT extends AbstractEsqlIntegTestCase {
                     && timeseries.getFirst().v2().v1().toEpochMilli() % (secondsInWindow * 1000L) == 0
                     && offset > 0) {
                     // Value at lower boundary is present, check if there's one in the previous window to use.
-                    // For INCREASE, return 0 because the increase was already accounted for in the previous bucket.
+                    addLastTupleFromLowerWindow(timeseries, allTimeseries.get(offset - 1), secondsInWindow);
+                    // For INCREASE, return 0 if there is a previous bucket because
+                    // the increase was already accounted for in the previous bucket.
                     // For RATE, we still need to calculate the rate using interpolation from the previous bucket.
-                    if (deltaAgg.equals(DeltaAgg.INCREASE)) {
+                    if (timeseries.size() == 2 && deltaAgg.equals(DeltaAgg.INCREASE)) {
                         return new RateRange(0.0, 0.0);
                     }
-                    addLastTupleFromLowerWindow(timeseries, allTimeseries.get(offset - 1), secondsInWindow);
                 }
                 if (timeseries.size() < 2) {
                     return null;

